app/jobs/resource_request_jobs/access_for_text_or_office_document_job.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module ResourceRequestJobs
+  class AccessForTextOrOfficeDocumentJob < ApplicationJob
+    include ResourceRequestJobs::ProcessingHelpers
+
+    queue_as :resource_request_access_for_text_or_office_document
+
+    # Creates an access copy for a text or office_document resource.
+    def perform(resource_request_id:, digital_object_uid:, src_file_location:, options: {})
+      with_shared_error_handling(resource_request_id) do
+        Rails.logger.info("Running #{self.class.name} job for resource request #{resource_request_id}")
+        validate_options!(options)
+
+        Hyacinth::Client.instance.resource_request_in_progress!(resource_request_id)
+        generate_and_upload(src_file_location, digital_object_uid, options)
+        Hyacinth::Client.instance.resource_request_success!(resource_request_id)
+      end
+    end
+
+    # Validates the given options, raising an Derivativo::Exceptions::OptionError if the options are invalid
+    def validate_options!(options)
+      validate_required_option!(options, 'format', ['pdf'])
+      true
+    end
+
+    private
+
+      def generate_and_upload(src_file_location, digital_object_uid, options)
+        file_prefix = 'access'
+        file_suffix = ".#{options['format']}"
+
+        # Reserve file in working directory to avoid name collisions with concurrent processes.
+        Derivativo::FileHelper.working_directory_temp_file(file_prefix, file_suffix) do |dst_file|
+          Derivativo::Conversion.text_or_office_document_to_pdf(src_file_path: Derivativo::FileHelper.file_location_to_file_path(src_file_location), dst_file_path: dst_file.path)
```

Comment:
No need to check in this case, since `text_or_office_document_to_pdf` doesn't have a significant return value.  It's a void return type method.  I like it when that's more explicit in other languages (like Java).